google-map-poly.html

```diff
@@ -0,0 +1,554 @@
+        this._createPoly();
+      }
+    },
+
+    _mouseEventsChanged: function() {
+      if (this.poly) {
+        if (this.mouseEvents) {
+          this._forwardEvent('mousedown');
+          this._forwardEvent('mousemove');
+          this._forwardEvent('mouseout');
+          this._forwardEvent('mouseover');
+          this._forwardEvent('mouseup');
+        } else {
+          this._clearListener('mousedown');
+          this._clearListener('mousemove');
+          this._clearListener('mouseout');
+          this._clearListener('mouseover');
+          this._clearListener('mouseup');
+        }
+      }
+    },
+
+    _strokeColorChanged: function() {
+      this.poly && this.poly.set('strokeColor', this.strokeColor);
+    },
+
+    _strokeOpacityChanged: function() {
+      this.poly && this.poly.set('strokeOpacity', this.strokeOpacity);
+    },
+
+    _strokePositionChanged: function() {
+      this.poly && this.poly.set('strokePosition', this._convertStrokePosition());
+    },
+
+    _strokeWeightChanged: function() {
+      this.poly && this.poly.set('strokeWeight', this.strokeWeight);
+    },
+
+    _zIndexChanged: function() {
+      this.poly && this.poly.set('zIndex', this.zIndex);
+    },
+
+    // Helper logic.
+
+    _buildPathFromPoints: function() {
+      this._points = Array.prototype.slice.call(Polymer.dom(this.$.points).getDistributedNodes());
+      window.console.log("Points:  " + this._points.length);
+      window.console.log(Polymer.dom(this).getDestinationInsertionPoints());
+
+      // Build path from current points (ignoring vertex insertions while doing so).
+      this._building = true;
+      this.path.clear();
+      for (var i = 0, point; point = this._points[i]; ++i) {
+        this.path.push(point.getPosition());
+      }
+      this._building = false;
+
+      this.fire('google-map-poly-path-built', this.path);
+
+      // Watch for future updates.
+      if (this._mutationObserver) {
+        return;
+      }
+      this._mutationObserver = new MutationObserver(this._buildPathFromPoints.bind(this));
+      this._mutationObserver.observe(this, {
+        childList: true
+      });
+    },
+
+    _clearListener: function(name) {
+      if (this._listeners[name]) {
+        google.maps.event.removeListener(this._listeners[name]);
+        this._listeners[name] = null;
+      }
+    },
+
+    _convertStrokePosition: function() {
+      return google.maps.StrokePosition && this.strokePosition?
+          google.maps.StrokePosition[this.strokePosition.toUpperCase()] : 0;
+    },
+
+    _createPoly: function() {
+      // Build poly's path and register mutation listeners on first creation.
+      if (!this.path) {
+        this._setPath(new google.maps.MVCArray());
+        google.maps.event.addListener(this.path, 'insert_at', this._startEditing.bind(this));
+        google.maps.event.addListener(this.path, 'set_at', this._updatePoint.bind(this));
+        this._buildPathFromPoints();
```

Comment:
seems like you could create the MO here and not have to deal with it in buildPathFromPoints. your comment above also describes it being here.